src/WeekView/WeekView.js

```diff
@@ -42,16 +45,28 @@ export default class WeekView extends Component {
   }
 
   componentDidUpdate() {
-    this.calendar.scrollTo({ y: 0, x: 2 * (SCREEN_WIDTH - 60), animated: false });
+    this.eventsGrid.scrollTo({ y: 0, x: 2 * (SCREEN_WIDTH - 60), animated: false });
+  }
+
+  scrollToAgendaStart = () => {
+    if (this.verticalAgenda) {
+      const { startHour, hoursInDisplay } = this.props;
+      const startHeight = startHour * CONTAINER_HEIGHT / hoursInDisplay;
+      this.verticalAgenda.scrollTo({ y: startHeight, x: 0, animated: false });
+    }
   }
 
   generateTimes = () => {
+    const { hoursInDisplay } = this.props;
     const times = [];
-    for (let i = 0; i < TIME_LABELS_COUNT; i += 1) {
-      const minutes = i % 2 === 0 ? '00' : '30';
-      const hour = Math.floor(i / 2);
-      const time = `${hour}:${minutes}`;
-      times.push(time);
+    const timeLabelsPerHour = TIME_LABELS_IN_DISPLAY / hoursInDisplay;
```

Comment:
This improvement is nice, but the props name `hoursInDisplay` is kind of confusing to me. Could we have another name?

Reply:
Sure! any suggestions? I can think of `hoursInScreen`, `displayedHours`, `agendaLength`, `agendaHours`, but they may be confusing as well.
Any other ideas are welcomed

Reply:
I can't think of a better name. We can improve later. Thanks again for the MR! I'm merging.